data/wsl/UpdateInstall.ps1

```diff
@@ -0,0 +1,80 @@
+# Define a function to log messages with timestamps
+function LogMessage {
+    param (
+        [string]$Message,
+        [string]$Color = "White"
+    )
+    $Timestamp = (Get-Date).ToString("yyyy-MM-dd HH:mm:ss")
+    Write-Host "$Timestamp - $Message" -ForegroundColor $Color
+}
+
+# Start logging
+LogMessage "Starting Windows Update process..."
+
+# Create Update Session
+$Session = New-Object -ComObject Microsoft.Update.Session
+$Searcher = $Session.CreateUpdateSearcher()
+
+# Search for Updates
+LogMessage "Searching for updates..."
+$SearchResult = $Searcher.Search("IsInstalled=0")
+
+# Check if updates are available
+if ($SearchResult.Updates.Count -eq 0) {
+    LogMessage "No updates found." -Color "Yellow"
+    return
+}
+
+# Display updates found
+LogMessage "$($SearchResult.Updates.Count) update(s) found."
+$UpdatesToInstall = New-Object -ComObject Microsoft.Update.UpdateColl
+foreach ($Update in $SearchResult.Updates) {
+    LogMessage "Update found: $($Update.Title)"
+    $UpdatesToInstall.Add($Update) | Out-Null
+}
+
+# Download updates individually
+LogMessage "Downloading updates..."
+foreach ($Update in $UpdatesToInstall) {
+    $Downloader = $Session.CreateUpdateDownloader()
+    $SingleUpdateColl = New-Object -ComObject Microsoft.Update.UpdateColl
+    $SingleUpdateColl.Add($Update) | Out-Null
+    $Downloader.Updates = $SingleUpdateColl
+    
+    LogMessage "Downloading: $($Update.Title)..."
+    try {
+        $DownloadResult = $Downloader.Download()
+        if ($DownloadResult.ResultCode -eq 2) {
+            LogMessage "Downloaded: $($Update.Title) successfully." -Color "Green"
+        } else {
+            LogMessage "Download failed for: $($Update.Title)" -Color "Yellow"
```

Comment:
Why is this yellow and not red? The download failed, this should also be treated as an error, no? :thinking: